x/commitment/keeper/vest.go

```diff
@@ -0,0 +1,53 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/elys-network/elys/x/commitment/types"
+)
+
+func (k Keeper) VestTokens(ctx sdk.Context) error {
+
+	k.IterateCommitments(ctx, func(commitments types.Commitments) (stop bool) {
+		logger := k.Logger(ctx)
+
+		for index := len(commitments.VestingTokens) - 1; index >= 0; index-- {
+			vesting := commitments.VestingTokens[index]
+			vesting.CurrentEpoch = vesting.CurrentEpoch + 1
+
+			epochAmount := vesting.TotalAmount.Quo(sdk.NewInt(vesting.NumEpochs))
+
+			withdrawCoins := sdk.NewCoins(sdk.NewCoin(vesting.Denom, epochAmount))
+
+			// Mint the vested tokens to the module account
+			err := k.bankKeeper.MintCoins(ctx, types.ModuleName, withdrawCoins)
+			if err != nil {
+				logger.Debug(
+					"unable to mint vested tokens",
+					"vestingtokens", vesting, commitments.Creator,
+				)
+			}
+			// Send the minted coins to the user's account
+			err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, sdk.AccAddress(commitments.Creator), withdrawCoins)
+			if err != nil {
+				logger.Debug(
+					"unable to send vested tokens",
+					"vestingtokens", vesting, commitments.Creator,
+				)
```

Comment:
I think in this case, it will need to use cacheCtx if the error is going to not stop the execution and it cause corrupt the bank keeper's invariant